src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3954,11 +3954,21 @@ private bool HasTopLevelNullabilityConversion(TypeWithAnnotations source, TypeWi
 
                 case ConversionKind.ExplicitDynamic:
                 case ConversionKind.ImplicitDynamic:
-                case ConversionKind.Unboxing:
                 case ConversionKind.Boxing:
                     resultState = operandType.State;
                     break;
 
+                case ConversionKind.Unboxing:
+                    if (operandType.MayBeNull && targetType.IsNonNullableValueType() && reportRemainingWarnings)
+                    {
+                        ReportSafetyDiagnostic(ErrorCode.WRN_UnboxPossibleNull, node.Syntax);
```

Comment:
Why do we not use the `operandType.State` here? In other cases of `reportRemainingWarnings` in this method we seem to prefer using it.

Reply:
If you're asking why we don't do `resultState = operandType.State`, it is because this is an operation that unboxes to a value type, which is never null.  We set `resultState` to `NotNull` before entering this switch, so we don't have to do anything more here.

---
In reply to: [273122384](https://github.com/dotnet/roslyn/pull/34781#discussion_r273122384) [](ancestors = 273122384)